src/dataflow-types/src/types.rs

```diff
@@ -742,6 +742,10 @@ pub enum SinkConnector {
 pub struct KafkaSinkConsistencyConnector {
     pub topic: String,
     pub schema_id: i32,
+    // gate_ts is the most recent high watermark tailed from the consistency topic
+    // Exactly-once sinks use this to determine when they should start publishing again. This
+    // tells them when they have caught up to where the previous materialize instance stopped.
+    pub gate_ts: Option<u64>,
```

Comment:
I think you should have this be a `Option<Timestamp>` and you can maybe also have this be just a `Timestamp` and set this value to `Timestamp::min()` in the absense of exactly once - iiuc that should do the same thing

Reply:
Good point! I went down the option path back when I had this all glommed together in kafka.rs, because I needed to distinguish there between "is not eos" and "is eos but has not retrieved timestamp yet". Now that all that logic moved it serves no purpose; much cleaner without 🙂

Reply:
Oof, I walked this back a little - UTs show that certain scenarios do still start with ts 0, and apart from that byo could theoretically do so, so min isn't a valid sentinel. I switched to using Timestamp, but keeping the Option to clearly disambiguate enabled vs enabled